pkg/reconciler/identity/reconciler.go

```diff
@@ -42,33 +41,38 @@ import (
 )
 
 const (
-	Add                          = "add"
-	Remove                       = "remove"
 	Role                         = "roles/iam.workloadIdentityUser"
 	deleteWorkloadIdentityFailed = "WorkloadIdentityDeleteFailed"
 	workloadIdentityFailed       = "WorkloadIdentityReconcileFailed"
-	concurrencyError             = "googleapi: Error 409: There were concurrent policy changes."
 )
 
-// defaultRetry represents that there will be 3 iterations.
-// The duration starts from 100ms and is multiplied by factor 2.0 for each iteration.
-var defaultRetry = wait.Backoff{
-	Steps:    5,
-	Duration: 100 * time.Millisecond,
-	Factor:   2.0,
-	// The sleep at each iteration is the duration plus an additional
-	// amount chosen uniformly at random from the interval between 0 and jitter*duration.
-	Jitter: 1.0,
-}
+// TODO remove global instance
+var (
+	policyManager       *IAMPolicyManager
+	createPolicyManager sync.Once
+)
 
 func NewIdentity(ctx context.Context) *Identity {
+	createPolicyManager.Do(func() {
+		c, err := admin.NewIamClient(context.Background())
+		if err != nil {
+			panic(err)
+		}
+		m, err := NewIAMPolicyManager(context.Background(), c)
+		if err != nil {
+			panic(err)
+		}
+		policyManager = m
+	})
 	return &Identity{
-		KubeClient: kubeclient.Get(ctx),
+		KubeClient:    kubeclient.Get(ctx),
+		PolicyManager: policyManager,
 	}
 }
 
 type Identity struct {
-	KubeClient kubernetes.Interface
+	KubeClient    kubernetes.Interface
+	PolicyManager *IAMPolicyManager
```

Comment:
Can this be private?

Reply:
I've left this public so that tests can construct instances using a fake client. What would probably be better would be to add it to the NewIdentity constructor and inject it somehow, but that requires figuring out how to do injection.

Reply:
I was able to refactor things to make this and KubeClient private